src/app/shop/item-filter/item-filter.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from '@angular/core';
+import { Item } from '../data';
+import { ItemsService } from '../items.service';
+import { Observable } from 'rxjs/internal/Observable';
+import { fromEvent } from 'rxjs';
+import { debounceTime } from 'rxjs/operators';
+
+@Component({
```

Comment:
Este componente no funciona, no filtra nada en la lista principal. Viendo el código está mal implementado, noté varias cosas:
- Para empezar no creo que ocupes un componente puesto que solo se pidió un input que cuando cambie filtre los items de la lista.